src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,222 @@
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt.prefSettingAdjuster");
+
+
+/*
+ * The following transform functions, valueInputToBinder and binderToValueInput
+ * sync the value being edited from our model to the adjust controler, and in
+ * this situation take in to account the semantics for binding boolean values by
+ * looking at the metadata schema of the term currently being edited, and if it
+ * is boolean applying that semantic.
+ *
+ * At the moment these are looking up the prefSettingAdjuster component manually,
+ * and should be changed to the correct way to set those values, and also prepare
+ * for a future where there is more than one instance of a pref adjuster at a time
+ * on a page.  They should really just be subcomponents on the prefSettingAdjuster
+ * component, and be able to look at the item currently being edited to apply the
+ * correct semantics. (Which could be even more complex in the future, at the moment
+ * we only handle very simple types.)
+ */
+
+fluid.defaults("gpii.devpmt.prefSettingAdjuster.valueInputToBinder", {
+    gradeNames: ["fluid.standardTransformFunction"]
+});
+
+gpii.devpmt.prefSettingAdjuster.valueInputToBinder = function (value) {
+    var adjuster = fluid.queryIoCSelector(fluid.rootComponent, "gpii.devpmt.prefSettingAdjuster")[0];
+    if (adjuster.model.metadata.schema.type === "boolean") {
+        return value ? ["on"] : [];
+    }
+    else {
+        return value;
+    }
+};
+
+fluid.defaults("gpii.devpmt.prefSettingAdjuster.binderToValueInput", {
+    gradeNames: ["fluid.standardTransformFunction"]
+});
+
+gpii.devpmt.prefSettingAdjuster.binderToValueInput = function (value) {
+    var adjuster = fluid.queryIoCSelector(fluid.rootComponent, "gpii.devpmt.prefSettingAdjuster")[0];
+    if (adjuster.model.metadata.schema.type === "boolean") {
+        return value.length > 0;
+    }
+    else {
+        return value;
+    }
+};
+
+/**
+ * A gpii handlebars widget to create the adjuster for the
+ * sidebar editing widget of the DevPMT.
+ *
+ * Each component will be constructed using the metadata
+ * object for the preference, which at a bare minimum should
+ * have a "title" entry and "schema" entry, the later which is
+ * a JSON schema that will be used to construct the widget
+ * layout and validate it's input.
+ *
+ * TODO A number of the html ID's and selectors on this need to be
+ * fixed up so that we can have more than one of them on a page
+ * at the same time.
+ */
+fluid.defaults("gpii.devpmt.prefSettingAdjuster", {
+    gradeNames: ["gpii.handlebars.templateAware", "gpii.binder.bindOnCreate", "gpii.binder.bindOnDomChange"],
```

Comment:
Again, `gpii.binder.onDomChange` is already provided by `gpii.handlebars.templateAware`.

Reply:
this was cleaned up